presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -168,6 +169,11 @@ private void validateShowStats(ShowStats node)
                 check(query.getQueryBody() instanceof QuerySpecification, node, "Only table and simple table subquery can be passed as argument to SHOW STATS clause");
                 QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
 
+                Optional<FilterNode> filterNode = searchFrom(plan.getRoot())
+                        .where(FilterNode.class::isInstance)
+                        .findSingle();
+
+                check(!filterNode.isPresent(), node, "Only predicates that can be pushed down are supported in the SHOW STATS WHERE clause");
```

Comment:
Would it make sense to inline all these checks? Or pass `Query` and `QuerySpecification` as arguments to avoid recomputing these?

Reply:
Agree, we can pass QuerySpecification to avoid duplicated cast